spectrumx_visualization_platform/spx_vis/api/views.py

```diff
@@ -1,119 +1,212 @@
+            capture_max_freq = float_check(capture.get("max_freq"))
+            capture_start_time = datetime_check(capture.get("timestamp"))
+            capture_end_time = datetime_check(capture.get("end_time"))
+
+            if min_freq is not None and capture_max_freq < min_freq:
+                return False
+            if max_freq is not None and capture_min_freq > max_freq:
+                return False
+            if start_time and (capture_end_time is None or capture_end_time < start_time):
+                return False
+            if end_time and (capture_start_time is None or capture_start_time > end_time):
+                return False
+
+            if source_filter and capture.get("source") != source_filter:
+                return False
+
+            return True
+
+        combined_file_list = list(filter(filter_capture, combined_file_list))
+
+    return Response({"captures": combined_file_list, "sds_count": sds_file_count})
+
+
+def calculate_end_time(start_time, scan_time):
+    print(f"scan time in end time func",scan_time)
+    start_time = datetime_check(start_time)
+    print(f"start_time in func calculate end time", start_time)
+    print(f"delta",timedelta(seconds=scan_time) )
+    if start_time and isinstance(scan_time, (int, float)):
+        print(f"delta",timedelta(seconds=scan_time) )
+        end_time = start_time + timedelta(seconds=scan_time)
+        print(f"Computed end_time: {end_time}")
+        return end_time.strftime("%Y-%m-%d %H:%M:%S.%f")  # Convert to string for JSON
+    print("Invalid timestamp or scan_time, returning None")
+    return None
+
+
+def datetime_check(value):
+    if not value:
+        return None
+    try:
+        return datetime.strptime(value, "%Y-%m-%dT%H:%M")
+    except ValueError:
+        try:
+            return datetime.strptime(value, "%Y-%m-%d %H:%M:%S.%f")
+        except ValueError:
+            return None
+
+
+def float_check(value, default=0.0):
+    """Safely convert a value to float."""
+    try:
+        return float(value)
+    except (ValueError, TypeError):
+        return default
 
 
 class CaptureViewSet(viewsets.ModelViewSet):
     queryset = Capture.objects.all()
     serializer_class = CaptureSerializer
     permission_classes = [permissions.IsAuthenticatedOrReadOnly]
+    print("i am here")
 
     def get_queryset(self):
-        """Get the queryset of captures for the current user.
-
-        Returns:
-            QuerySet: Filtered queryset containing only the user's captures.
-        """
         return Capture.objects.filter(owner=self.request.user)
 
     def create(self, request, *args, **kwargs):
-        """Create a new capture or captures.
-
-        For RadioHound captures, creates multiple captures (one per file).
-        For other types, creates a single capture with multiple files.
-
-        Returns:
-            Response: Created capture(s) data with appropriate status code
-        """
```

Comment:
I think we should keep most of these comments that are deleted here and in the rest of the file. Also, another leftover debugging print statement above